Assets/MRTK/SDK/Experimental/Elastic/Scripts/QuaternionElasticSystem.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Runtime.CompilerServices;
+using UnityEngine;
+
+[assembly: InternalsVisibleTo("Microsoft.MixedReality.Toolkit.Tests.PlayModeTests")]
+namespace Microsoft.MixedReality.Toolkit.Experimental.Physics
+{
+    public class QuaternionElasticSystem : ElasticSystem<Quaternion>
+    {
+        private Quaternion currentValue;
+        private Quaternion currentVelocity;
+
+        private QuaternionElasticExtent extent;
+        private ElasticProperties elasticProperties;
+
+        public QuaternionElasticSystem(Quaternion initialValue,
+                                       Quaternion initialVelocity,
+                                       QuaternionElasticExtent extentInfo,
+                                       ElasticProperties elasticProperties)
+        {
+            currentValue = initialValue;
+            currentVelocity = initialVelocity;
+            this.extent = extentInfo;
+            this.elasticProperties = elasticProperties;
+        }
+
+        /// <inheritdoc/>
+        public override Quaternion ComputeIteration(Quaternion forcingValue, float deltaTime)
+        {
+            if (Quaternion.Dot(forcingValue, currentValue) < 0)
+            {
+                forcingValue = new Quaternion(-forcingValue.x, -forcingValue.y, -forcingValue.z, -forcingValue.w);
+            }
+
+            // For clarity and conciseness
+            var k = elasticProperties.HandK;
```

Comment:
Nit: In general, avoid one letter variable names unless you're doing something like for (int i = 0....)

The difference between being able to read later down the function (1 / mass) vs (1 / m) is worth the additional 3-5 characters (i.e. not having to read up the function to be like - what does m stand for again?)

I think this mostly applies for drag/mass, less so for HandK (where K can be thought of as a "constant"/thing)

Reply:
I typically don't use any one letter variables either, but given that this is really just a mathematical evaluation of a formula it really does help clarity to have single-letter variables. I can change it though.

Reply:
Fixed